plugins/auth/src/components/ConsentPage/ConsentPage.tsx

```diff
@@ -0,0 +1,348 @@
+          window.location.href = result.redirectUrl;
+        }
+      } catch (err) {
+        alertApi.post({
+          message: isError(err) ? err.message : `Failed to ${action} consent`,
+          severity: 'error',
+        });
+      } finally {
+        setSubmitting(false);
+      }
+    },
+    [session, discoveryApi, fetchApi, alertApi],
+  );
+
+  if (!sessionId) {
+    return (
+      <Page themeId="tool">
+        <Header title="Authorization Error" />
+        <Content>
+          <EmptyState
+            missing="data"
+            title="Invalid Request"
+            description="The consent request ID is missing or invalid."
+          />
+        </Content>
+      </Page>
+    );
+  }
+
+  if (loading) {
+    return (
+      <Page themeId="tool">
+        <Header title="Authorization Request" />
+        <Content>
+          <Box
+            display="flex"
+            justifyContent="center"
+            alignItems="center"
+            minHeight={300}
+          >
+            <Progress />
+          </Box>
+        </Content>
+      </Page>
+    );
+  }
+
+  if (error ?? !session) {
+    return (
+      <Page themeId="tool">
+        <Header title="Authorization Error" />
+        <Content>
+          <ResponseErrorPanel
+            error={new Error(error || 'Authorization request not found')}
+          />
+        </Content>
+      </Page>
+    );
+  }
+
+  if (completed) {
+    return (
+      <Page themeId="tool">
+        <Header title="Authorization Complete" />
+        <Content>
+          <Card className={classes.authCard}>
+            <CardContent>
+              <Box textAlign="center">
+                {completed.action === 'approve' ? (
+                  <CheckCircleIcon
+                    style={{ fontSize: 64, color: 'green', marginBottom: 16 }}
+                  />
+                ) : (
+                  <CancelIcon
+                    style={{ fontSize: 64, color: 'red', marginBottom: 16 }}
+                  />
+                )}
+                <Typography variant="h5" gutterBottom>
+                  {completed.action === 'approve'
+                    ? 'Authorization Approved'
+                    : 'Authorization Denied'}
+                </Typography>
+                <Typography variant="body1" color="textSecondary" gutterBottom>
+                  {completed.action === 'approve'
+                    ? 'You have successfully authorized the application to access your Backstage account.'
+                    : 'You have denied the application access to your Backstage account.'}
+                </Typography>
+                <Typography variant="body2" color="textSecondary">
+                  Redirecting to the application...
+                </Typography>
+              </Box>
+            </CardContent>
+          </Card>
+        </Content>
+      </Page>
+    );
+  }
+
+  const appName = session.clientName ?? session.clientId;
+
+  return (
+    <Page themeId="tool">
```

Comment:
maybe with a wrapping component you could have let an outer component render the page and header and content root (less repetition) and the inner one do the flow